### PR TITLE
Request a session with 'hit-test' in the list of requiredFeatures

### PR DESCRIPTION
**Pull request by bialpio**
Ensure that the session is created with `hit-test` listed as required feature, other minor changes.

### CODE REVIEW

File: hit-test.html [hunk 1]
```diff
@@ -57,6 +57,8 @@
       let xrViewerSpace = null;
       let xrHitTestSource = null;
 
+      const sessionInit = {requiredFeatures: ['local', 'hit-test']};
```
Comment (toji) [inline hit-test.html]: 'local' should always implicitly be a required feature for immersive sessions. If this was necessary to get the page to run properly that's a bug.

File: hit-test.html [hunk 2]
```diff
@@ -97,18 +99,19 @@
         document.querySelector('header').appendChild(xrButton.domElement);
 
         if (navigator.xr) {
-          navigator.xr.isSessionSupported('immersive-ar').then((supported) => {
+          navigator.xr.isSessionSupported('immersive-ar', sessionInit)
```
Comment (toji) [inline hit-test.html]: `isSessionSupported()` shouldn't take any options other than the session mode.

File: hit-test.html [hunk 3]
```diff
@@ -146,6 +149,7 @@
       }
 
       function onEndSession(session) {
+        xrHitTestSource.cancel();
```
Comment (toji) [inline hit-test.html]: Curious as to why this is beneficial? I'd imagine that the hit test sources don't do anything once the session is ended?

Comment (bialpio) [reply hit-test.html]: Mostly educational - I do not want people to just drop all references w/o canceling hit test sources. In this particular case it does not really matter.

File: hit-test.html (hunk 1 above)
Comment (bialpio) [reply hit-test.html]: It worked fine w/o it but I believe that explicit > implicit. Happy to remove it if you feel strongly about it, I'm not sure what the conventions are.

File: hit-test.html (hunk 2 above)
Comment (bialpio) [reply hit-test.html]: D'oh! Not sure why I thought it does...

**Pull request merged**